From ce4a2ba118ae26322f86ab2b3cddc8d44c700af8 Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Thu, 14 Jul 2016 16:05:52 -0400
Subject: [PATCH 0649/1722] drm/amd/dal: fix pplib min sclk reporting for dce80

For asics that does not have clk levels we just report the actual
required sclk

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index 5fb62a13..5c33365 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -714,6 +714,13 @@ static int determine_sclk_from_bounding_box(
 {
 	int i;
 
+	/*
+	 * Some asics do not give us sclk levels, so we just report the actual
+	 * required sclk
+	 */
+	if (dc->sclk_lvls.num_levels == 0)
+		return required_sclk;
+
 	for (i = 0; i < dc->sclk_lvls.num_levels; i++) {
 		if (dc->sclk_lvls.clocks_in_khz[i] >= required_sclk)
 			return dc->sclk_lvls.clocks_in_khz[i];
@@ -726,6 +733,7 @@ static int determine_sclk_from_bounding_box(
 	ASSERT(0);
 	return dc->sclk_lvls.clocks_in_khz[dc->sclk_lvls.num_levels - 1];
 }
+
 void pplib_apply_display_requirements(
 	const struct core_dc *dc,
 	const struct validate_context *context,
-- 
2.7.4

